spec/sms_tools/encoding_detection_spec.rb

```diff
@@ -95,6 +117,26 @@
       detection_for('Уникод: ^{}[~]|€\\').length.must_equal 17
       detection_for('Уникод: Σ: €').length.must_equal 12
     end
+
+    describe 'with SmsTools.use_gsm_encoding = false' do
+      before do
+        SmsTools.use_gsm_encoding = false
+      end
+
+      it "returns ASCII encoded length for some specific symbols which are also in GSM 03.38" do
+        detection_for('[]').length.must_equal 2
```

Comment:
I'm slightly concerned that we're enshrining what I'm pretty sure is a bug in this spec, but at least we're documenting current behaviour! Might be worth adding a comment stating this actually might not be the behaviour we want, given no SMS service will actually use ASCII encoding.